packages/graphql-eslint-rules/src/missing-apollo-key-fields.ts

```diff
@@ -281,17 +278,15 @@ const missingApolloKeyFieldsRule: GraphQLESLintRule<
                     },
                   },
                 };
+                const checkedFragments =
+                  checkedFragmentSpreads.size === 0
+                    ? ""
+                    : `(${Array.from(checkedFragmentSpreads).join(", ")})`;
+                const fieldName = unusedKeyFields.join(", ");
 
                 context.report({
                   node: newNode,
-                  messageId: REQUIRE_KEY_FIELDS_WHEN_AVAILABLE,
-                  data: {
-                    checkedFragments:
-                      checkedFragmentSpreads.size === 0
-                        ? ""
-                        : `(${Array.from(checkedFragmentSpreads).join(", ")})`,
-                    fieldName: unusedKeyFields.join(", "),
-                  },
+                  message: `Field(s) "${fieldName}" must be selected (when available on a type). Please make sure to include it in your selection set!\nIf you are using fragments, make sure that all used fragments ${checkedFragments} specifies the field(s) "${fieldName}".`,
```

Comment:
I would suppose that each document (i.e. fragment) gets checked, so no need to list all the fragments in each error message. (Also did some sprucing up of the message formatting.)

```suggestion
                  message: `The key-field${unusedKeyFields.length === 1 ? "" : "s"} ${unusedKeyFields.length === 1 ? unusedKeyFields[0] : unusedKeyFields.slice(0, -1).join(", ") + " and " + unusedKeyFields[unusedKeyFields.length - 1]} must be selected for proper Apollo Client store denormalisation purposes.`,
```